src/lib/components/dropzone/Dropzone.tsx

```diff
@@ -59,9 +59,17 @@ export function Dropzone({
     useDropzone({
       ...dropzoneOptions,
       onDrop: (acceptedFiles, rejectedFiles, e) => {
+        const { multiple } = dropzoneOptions;
+
         setAcceptedFiles((alreadyAcceptedFiles) => {
-          const newAcceptedFiles = [...alreadyAcceptedFiles, ...acceptedFiles];
-          if (onChange) {
+          const singleFileAlreadyHasFile =
+            !multiple && alreadyAcceptedFiles.length === 1;
+
+          const newAcceptedFiles = singleFileAlreadyHasFile
+            ? alreadyAcceptedFiles
+            : [...alreadyAcceptedFiles, ...acceptedFiles];
```

Comment:
I'm not sure we that we should change the signature of onChange when using a non multiple dropzone. I think it is fine to keep `onChange(files[])` even if the files array is constrained to a single entry.
The reasoning behind that is multiple:
 - We may imagine an use case where a toggle change a single -selectable dropzone to a multiple one and if the signature of the onChange method is different for both it would complexify the integration of the dropzone
 -  Making this signature change complexifies the typing of the dropzone